### DISCUSSION TITLE
ESP-01 Relay Board Update code

### DISCUSSION THREAD

**Discussion by xhang1108** [RavenSystem/esp-homekit-devices#2092] [MEPLHAA Scripts]
My old code is using this, I try to convert to new code but not working.

{"c":{"1":1,"b":[{"g":2,"t":5}],"a":[{"t":1,"s":1,"0":{"r":[{"g":0,"v":1}]},"1":{"r":[{"g":0,"v":0}]},"b":[{"g":2}]}]}

**Comment by Tellicious** [comment]
Try this:
Button connected to GPIO2, relay connected to GPIO0, relay OFF with GPIO0 HIGH, relay ON with GPIO0 LOW. Initial status GPIO0 HIGH -> relay OFF

`{"c":{"io":[[[0],2],[[2],6]],"b":[[2,5]]},"a":[{"0":{"r":[[0,1]]},"1":{"r":[[0]]},"b":[[2]],"s":1}]}`

**Reply by shengweitw** [reply]
May I ask where to convert the old and new JSON CODE?

**Reply by Tellicious** [reply]
AFAIK there isn't a dedicated tool to do so. The only major change is related to GPIO definition. You should start from the Wiki

**Reply by xhang1108** [reply]
@Tellicious Thanks buddy, it's worked!
